components/kyma-metrics-collector/pkg/edp/client_test.go

```diff
@@ -105,19 +117,25 @@ func TestClientRetry(t *testing.T) {
 	g.Expect(err).Should(gomega.BeNil())
 	g.Expect(gotReq.URL.Host).To(gomega.Equal(edpURL.Host))
 
+	// when
 	_, err = edpClient.Send(gotReq, testData)
+
+	// then
 	g.Expect(err).ShouldNot(gomega.BeNil())
 	g.Expect(err.Error()).Should(gomega.Equal("failed to POST event to EDP: failed to send event stream as EDP returned HTTP: 500"))
 	g.Expect(countRetry).Should(gomega.Equal(expectedCountRetry))
 
-	// Ensure metric exists
-	g.Expect(testutil.CollectAndCount(totalRequest, metricsName)).Should(gomega.Equal(1))
-	g.Expect(testutil.CollectAndCount(sentRequestDuration, histogramName)).Should(gomega.Equal(1))
+	// ensure metric exists.
+	g.Expect(testutil.CollectAndCount(latencyMetric, histogramName)).Should(gomega.Equal(1))
 
-	// Ensure metric has expected value
-	status500Counter, err := totalRequest.GetMetricWithLabelValues(fmt.Sprint(http.StatusInternalServerError))
+	// ensure metric has expected label value.
+	pMetric, err := kmctesting.PrometheusGatherAndReturn(latencyMetric, histogramName)
 	g.Expect(err).Should(gomega.BeNil())
-	g.Expect(testutil.ToFloat64(status500Counter)).Should(gomega.Equal(float64(1)))
+	g.Expect(pMetric.Metric).Should(gomega.HaveLen(1))
+	g.Expect(pMetric.Metric[0].Label).Should(gomega.HaveLen(2))
```

Comment:
If you use `pMetric.Metric[0].Label` twice you could just store it in a var with a descriptive name.

Reply:
updated.